In [46]:
from sklearn import linear_model
from sklearn.metrics import *
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
import pandas 


We will use the dataset provided in the path to make the prediction of the price. 

In [47]:
path = 'https://raw.githubusercontent.com/speianudana/FIA/main/Lab3/apartmentComplexData.txt'
columns = ['1', '2', 'complex_age', 'total_rooms', 'total_bedrooms', 'complex_inhabitants', 'apartments_nr', '8', 'median_complex_value'] 


Then using the pandas library we can work with data in a table manner. So, reading from a text file the Pandas library will create a data frame that is in a table form.

In [48]:
dataset = pandas.read_csv(path, names=columns)
dataset.describe()

1             2  ...             8  median_complex_value
count  20640.000000  20640.000000  ...  20640.000000          20640.000000
mean    -119.569704     35.631861  ...      3.870671         206855.816909
std        2.003532      2.135952  ...      1.899822         115395.615874
min     -124.350000     32.540000  ...      0.499900          14999.000000
25%     -121.800000     33.930000  ...      2.563400         119600.000000
50%     -118.490000     34.260000  ...      3.534800         179700.000000
75%     -118.010000     37.710000  ...      4.743250         264725.000000
max     -114.310000     41.950000  ...     15.000100         500001.000000

[8 rows x 9 columns]

Let's see which columns has the biggest influence on the median_complex_value using the correlation.

In [49]:
dataset.corr()


1         2  ...         8  median_complex_value
1                     1.000000 -0.924664  ... -0.015176             -0.045967
2                    -0.924664  1.000000  ... -0.079809             -0.144160
complex_age          -0.108197  0.011173  ... -0.119034              0.105623
total_rooms           0.044568 -0.036100  ...  0.198050              0.134153
total_bedrooms        0.068378 -0.066318  ... -0.008093              0.050594
complex_inhabitants   0.099773 -0.108785  ...  0.004834             -0.024650
apartments_nr         0.055310 -0.071035  ...  0.013033              0.065843
8                    -0.015176 -0.079809  ...  1.000000              0.688075
median_complex_value -0.045967 -0.144160  ...  0.688075              1.000000

[9 rows x 9 columns]

Here we observe that the biggest correlation has the column 8 as we can observe than its value is the biggest - 0.688075.

Next we will need to spil the data into data used for training the model, and data used for testing it.

In [50]:
x =  data_for_training[['1', '2', 'complex_age','total_rooms','total_bedrooms', 'complex_inhabitants','apartments_nr','8']]

y =  data_for_training[['median_complex_value']].values.flatten()


In [ ]:
The training data size is 80% and the testing data size is 20% of all dataset.

In [51]:
x_train, x_test, y_train, y_test = train_test_split(x, y, train_size=80)

Then we will use the LinearRegression().fit() method will train the model using the x_train -that is training data , and y_train - traget values.

In [52]:
training = LinearRegression().fit(x_train, y_train)
training


LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

Below we can see the score for training model where x_train - test samples, y_train - true values. And the score for testing model for testing the training model the same: x_test- test samples and y_test - true values. 



In [53]:
print("Training score:", training.score(x_train, y_train))
print("Test score:", training.score(x_test, y_test))



Training score: 0.6784214015849696
Test score: 0.6108275661439804


Below are the results for the actual price and the predicted price by our model.

In [54]:
print("The actual price:", y_test[0])
print("Predicted price:", training.predict([x_test.iloc[0]])[0])


The actual price: 144800.0
Predicted price: 212410.94266386796
